bond_calculator.py

```diff
@@ -115,7 +112,14 @@ def calc_macaulay_duration(self, bond, yld):
         time to cashflow weighted by PV
         """
         # TODO: implement details here - Ching Kung
+        # wavg weighted average term to maturity
+        wavg = ((1 + yld) / (bond.payment_freq * yld) - ((1+ yld + bond.term * (bond.coupon - yld))))

```

Comment:
These code contains errors like `TypeError: unsupported operand type(s) for *: 'PaymentFrequency' and 'float'` because `payment_freq` is `enum`.

Also, you may not want to hard code the yld into this calculation because different bond could have different periodicity, discount factor is calculated by function `calc_one_period_discount_factor`